### PR TITLE
prov/gni: add domain op for toggling xpmem on/off

### PR DESCRIPTION
**Pull request by hppritcha**
Signed-off-by: Howard Pritchard howardp@lanl.gov

### CODE REVIEW

**Comment by hppritcha** [comment]
forgot to at someone
@sungeunchoi

**Comment by hppritcha** [comment]
related to #850

File: prov/gni/src/gnix_dom.c [hunk 1]
```diff
@@ -354,6 +354,9 @@ __gnix_dom_ops_get_val(struct fid *fid, dom_ops_val_t t, void *val)
 	case GNI_MR_HARD_STALE_REG_LIMIT:
 		*(int32_t *)val = domain->mr_cache_attr.hard_stale_limit;
 		break;
+	case GNI_XPMEM_ENABLE:
+		*(bool *)val = domain->params.xpmem_enabled;
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_dom.c:358]: Hmm...  should this maybe warn GNI_XPMEM_ENABLE is not set?

File: prov/gni/src/gnix_dom.c [hunk 2]
```diff
@@ -472,6 +475,11 @@ __gnix_dom_ops_set_val(struct fid *fid, dom_ops_val_t t, void *val)
 			return -FI_EINVAL;
 		domain->udreg_reg_limit = *(int32_t *) val;
 		break;
+	case GNI_XPMEM_ENABLE:
+#if HAVE_XPMEM
+		domain->params.xpmem_enabled = *(bool *)val;
+#endif
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_dom.c:488]: Ditto here.  Should we warn?

**Comment by hppritcha** [comment]
pushed updated commit per PR feedback

**Comment by sungeunchoi** [comment]
👍

**Pull request merged**